examples/employee-app/src/main/java/com/example/helidon/employee/EmployeeService.java

```diff
@@ -0,0 +1,164 @@
+     */
+    @Override
+    public void update(Routing.Rules rules) {
+        rules.get("/", this::getAll)
+            .get("/lastname/{name}", this::getByLastName)
+            .get("/department/{name}", this::getByDepartment)
+            .get("/title/{name}", this::getByTitle)
+            .post("/", this::save)
+            .get("/{id}", this::getEmployeeById)
+            .put("/{id}", this::update)
+            .delete("/{id}", this::delete);
+        LOGGER.info("Driver "+dbCreds.getProperty("drivertype"));
+    }
+
+    private void getAll(final ServerRequest request, final ServerResponse response) {
+    	List<Employee> allEmployees = this.employees.getAll();
+        response.send(allEmployees);
+    }
+
+    private void getByLastName(final ServerRequest request, final ServerResponse response) {
+        
+        // Invalid query strings handled in isValidQueryStr. Keeping DRY
+        if (isValidQueryStr(response, request.path().param("name"))){
+            response.status(200).send(this.employees.getByLastName(request.path().param("name")));
+        }
+    }
+
+    private void getByTitle(final ServerRequest request, final ServerResponse response) {
+        
+        if (isValidQueryStr(response, request.path().param("name"))){
+            response.status(200).send(
+                this.employees.getByTitle(request.path().param("name")));
+        }
+    }
+
+    private void getByDepartment(final ServerRequest request, final ServerResponse response) {
+
+        if (isValidQueryStr(response, request.path().param("name"))){
+            response.status(200).send(
+                this.employees.getByDepartment(request.path().param("name")));
+        }
+    }
+
+    private void getEmployeeById(ServerRequest request, ServerResponse response) {
+        
+        // If invalid, response handled in isValidId. Keeping DRY
+        if (isValidId(response, request.path().param("id"))){        
+            Employee employee = this.employees.getById(request.path().param("id"));
+            response.status(200).send(employee);                
+        }        
+    }
+
+    /* This is add/create */
+    private void save(ServerRequest request, ServerResponse response) {   	 
+        request.content().as(Employee.class)
+            .thenApply(e -> Employee.of(e.getFirstName(), e.getLastName(), e.getEmail(), e.getPhone(),
+            e.getBirthDate(), e.getTitle(), e.getDepartment()))
+            .thenApply(this.employees::save)
+            .thenCompose(p -> {
+                response.status(201).headers().location(URI.create("/employees/" + p.getId()));
+                return response.send();
+            });
+    }
+
+    private void update(ServerRequest request, ServerResponse response) { 
+
+        if (isValidId(response, request.path().param("id"))){
+            request.content().as(Employee.class)
+                .thenApply(e -> {
+                    tempEmployee.setFirstName(e.getFirstName());

```

Comment:
Instead of updating the same instance, you can just build a new one using `Employee.of`.
This would avoid the need for setter methods and would allow the fields of Employee to be immutable.

Reply:
This should be fixed now.